### PR TITLE
test: MRE moved to integration_tests

### LINKED ISSUE #4948: move test to integration test
https://github.com/lfortran/lfortran/pull/4915/files#r1783634622

@jinangshah21 please address this.

### PR DESCRIPTION
**Pull request by jinangshah21**
Closes #4948

This PR Addresses [Issue #4948](https://github.com/lfortran/lfortran/issues/4948) (moved MRE into integration tests)

### CODE REVIEW

**Review by kmr-srbh — CHANGES_REQUESTED** [review]
@jinangshah21 please register this test inside `integration_tests/CMakeLists.txt`, then remove it from `tests/tests.toml`. After making the above changes, please delete the created reference files. Ideally, as this test does not throw an error and the ASR output is not required for future use, we should not have this as a reference test.

- Reference tests -> for keeping a reference of error messages or AST/ASR/LLVM output.
- Integration tests -> general testing of features and fixes.

**Comment by jinangshah21** [comment]
This test is generating error for llvm, So till ASR is only true for this test. But moving it to CMakeLists will throw error for integration tests.

**Comment by kmr-srbh** [comment]
> This test is generating error for llvm, So till ASR is only true for this test. But moving it to CMakeLists will throw error for integration tests.

Could you please post the error you get? If we have an error, then it is a bug we need to fix.

**Comment by jinangshah21** [comment]
> > This test is generating error for llvm, So till ASR is only true for this test. But moving it to CMakeLists will throw error for integration tests.
> 
> Could you please post the error you get? If we have an error, then it is a bug we need to fix.
This is [llvm error](https://github.com/lfortran/lfortran/pull/4915#issuecomment-2383386423) for this test.

File: tests/tests.toml [hunk 1]
```diff
@@ -4210,5 +4210,5 @@ filename = "errors/func_parameter_type.f90"
 asr = true
 
 [[test]]
-filename = "errors/func_parameter_type_02.f90"
+filename = "../integration_tests/func_parameter_type_02.f90"
 asr_implicit_interface_and_typing = true
```
[the remark this answers is not in the document]
Comment (jinangshah21) [reply tests/tests.toml:4214]: But it is giving error for llvm so I think we can keep reference test of ASR in lfortran and register in integration tests with gfortran.

**Review by Pranavchiku — APPROVED** [review]
The test does not work with latest main. It just generates ASR.

**Comment by certik** [comment]
Register it and run it with gfortran at least, if we can't compile with lfortran yet.

**Review by certik — APPROVED** [review]

**Pull request merged**